interview-questions.md

```diff
@@ -8,42 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is object-oriented programming? How is it different than functional programming?
 
-Your answer:
+Your answer: OOP is everything in the "program" is an object which is an instance of a class. Functional programming is basically working solely with functions and calling your method within a function. OOP and functional programming are different because methods you can perform method right on the class of objects, as functional programming you would called the method within the function.
 
-Researched answer:
+Researched answer: OOP is data encapsulation and inheritence focused. Each class defines its own behaior attributes of instances. Functions can be passes as arguments to other functions.
 
-1. What is the difference between a Float and an Integer in Ruby?
+2. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: Intergers a whole number as floats are decimal numbers.
 
-Researched answer:
+Researched answer: Intergers a whole number and have integer operators such as \*, +, -, as floats are reserved for decimal numbers such as doing equations like 6.2 / 3.
 
-1. Ruby has an implicit return. What does that mean?
+3. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: Ruby will return the last line without the need of inserting a return.
 
-Researched answer:
+Researched answer: It is reported that an implicit return in Ruby will return the last line of code automatically. Only if the code is one line. To be able to have less code.
 
-1. What is a block in Ruby?
+4. What is a block in Ruby?
 
-Your answer:
+Your answer: Blocks are code functionality that gets passed to a method. Defined with a "Do" and an "End"
 
-Researched answer:
+Researched answer: Block is a anonymous function thats passed to a method to determine the ourcome.
+Ex.
+my_name = DAndre
 
-1. How do you extract a value in a Ruby hash?
+my_name.reverse do |value|
+p "my_name"
+end
 
```

Comment:
Try to run this example in a Ruby file and see what comes out. I see the idea, but make sure things are doing what you are expecting them to! But I love that you used an example in your interview answer!